### PR TITLE
upgrade to Go 1.19 - upgrade go.mod file & replace deprecated calls

### PR DESCRIPTION
**Pull request by jzhn**
*Issue #, if available:*
None

*Description of changes:*
Upgrade eks-connector in preparation of golang 1.19 build.

- Replaced deprecated usage of [io/ioutils](https://pkg.go.dev/io/ioutil)
- Updated go.mod version to 1.19

*Tests:*

- Existing unit tests in project
- Manual e2e test in EKS 1.21 cluster.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,3 +16,45 @@ require (
 	k8s.io/client-go v0.21.2
 	k8s.io/klog/v2 v2.8.0
 )
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
```
Comment (dariobritos) [inline go.mod:21]: Maybe a 
BEGIN Indirect
END Indirect
would be better than the comment on all the dependencies.

**Comment by dariobritos** [comment]
So I see 3 main changes:
Dependency changes
io/ioutil -> os
And vendor file deleted

Can you explain the last one?

**Review by lynnnnnnluo — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (jzhn) [reply go.mod:21]: This file is auto generated by `go mod tidy`. It is not supposed to be modified by hand.

**Comment by jzhn** [comment]
> And vendor file deleted

The go.mod/go.sum files deletion in vendor modules are done by rerunning `go mod vendor`. This change is documented in [go1.17 doc](https://go.dev/doc/go1.17#vendor)

>  If the main module specifies go 1.17 or higher, go mod vendor now omits go.mod and go.sum files for vendored dependencies, which can otherwise interfere with the ability of the go command to identify the correct module root when invoked within the vendor tree.

**Review by dariobritos — APPROVED** [review]

**Pull request merged**